editor/src/utils/utils.ts

```diff
@@ -920,9 +924,23 @@ export function isOptionsType<T extends OptionTypeBase>(
   return Array.isArray(value)
 }
 
+function deduplicateBy<T>(key: (t: T) => string, ts: Array<T>): Array<T> {
+  const seen = new Set<string>()
+  const results: Array<T> = []
+  for (const t of ts) {
+    const k = key(t)
+    if (!seen.has(k)) {
+      results.push(t)
+      seen.add(k)
+    }
+  }
+  return results
+}
```

Comment:
Nice add, but we already have a function for exactly this, called `uniqBy` in `src/core/shared/array-utils.ts` 😁